presto-main/src/main/java/com/facebook/presto/operator/scalar/MapConstructor.java

```diff
@@ -140,6 +150,10 @@ public static Block createMap(
                 mapBlockBuilder.closeEntry();
                 throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "map key cannot be null");
             }
+            if (isIndeterminate(keyIndeterminate, mapType.getKeyType(), keyBlock, i)) {
+                mapBlockBuilder.closeEntry();
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "map key cannot be indeterminate");
```

Comment:
Use `type.getObjectValue` to print the indeterminate in the error message to help user debugging:

```java
throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "map key cannot be indeterminate: " + mapType.getvalueType.getObjectValue(session, keyBlock, i));
```